### PR TITLE
JBPM-5604 Subform rename or its new copy is not reflected in the parent form

### PR DESCRIPTION
**Pull request by alessioscalici**
**WARNING** this is the same as https://github.com/kiegroup/kie-wb-common/pull/1152, pointing to master branch

@pefernan @jsoltes could you please review?

### CODE REVIEW

File: kie-wb-common-forms/kie-wb-common-forms-editor/kie-wb-common-forms-editor-client/src/main/java/org/kie/workbench/common/forms/editor/client/editor/FormEditorPresenter.java [hunk 1]
```diff
@@ -267,6 +270,58 @@ public void execute() {
...
+            renamePopupView.hide();
+            view.hideBusyIndicator();
+            notification.fire(new NotificationEvent(org.kie.workbench.common.widgets.client.resources.i18n.CommonConstants.INSTANCE.ItemRenamedSuccessfully(),
+                                                    NotificationEvent.NotificationType.SUCCESS));
+            doLoadContent(content);
+        };
+    }
+
+    protected DefaultErrorCallback getRenameErrorCallback(final RenamePopUpPresenter.View renamePopupView) {
+        return new DefaultErrorCallback() {
```
Comment (jhrcek) [inline kie-wb-common-forms/kie-wb-common-forms-editor/kie-wb-common-forms-editor-client/src/main/java/org/kie/workbench/common/forms/editor/client/editor/FormEditorPresenter.java]: There's `HasBusyIndicatorDefaultErrorCallback` in the same package `DefaultErrorCallback` that you could use to save some keystrokes here..

File: kie-wb-common-forms/kie-wb-common-forms-editor/kie-wb-common-forms-editor-backend/src/test/java/org/kie/workbench/common/forms/editor/backend/service/impl/FormEditorServiceImplTest.java [hunk 2]
```diff
@@ -109,7 +113,8 @@ public void init() {
                                                       formDefinitionSerializer,
                                                       vfsFormFinderService,
                                                       deleteService,
-                                                      commentedOptionFactory);
+                                                      commentedOptionFactory,
+                                                      renameService);
 
         when(path.toURI()).thenReturn("default:///src/main/resources/test.frm");
```
Comment (jhrcek) [inline kie-wb-common-forms/kie-wb-common-forms-editor/kie-wb-common-forms-editor-backend/src/test/java/org/kie/workbench/common/forms/editor/backend/service/impl/FormEditorServiceImplTest.java]: Could you please add a few tests for this new rename method of FormEditorService?

**Comment by jsoltes** [comment]
@alessioscalici I am back from vacation. will check it once the failures are fixed.

**Comment by manstis** [comment]
@alessioscalici You really need to see these PRs through to completion!

**Comment by pefernan** [comment]
@manstis I have @alessioscalici working hard on other stuff ;)... he'll fix everything soon.

**Comment by alessioscalici** [comment]
@jhrcek @pefernan @manstis @jsoltes I have addressed the comments

**Comment by pefernan** [comment]
jenkins execute full downstream build

File: kie-wb-common-forms/kie-wb-common-forms-editor/kie-wb-common-forms-editor-client/src/main/java/org/kie/workbench/common/forms/editor/client/editor/FormEditorPresenter.java [hunk 3]
```diff
@@ -268,6 +254,64 @@ public void execute() {
...
+            view.showSavePopup(versionRecordManager.getCurrentPath(),
+                               () -> rename(true),
+                               () -> rename(false));
+        } else {
+            rename(false);
+        }
+    }
+
+    public void rename(boolean save) {
+
```
Comment (pefernan) [inline kie-wb-common-forms/kie-wb-common-forms-editor/kie-wb-common-forms-editor-client/src/main/java/org/kie/workbench/common/forms/editor/client/editor/FormEditorPresenter.java]: There's an issue that when the form is dirty and the user presses the Save the form editor isn't reloaded well. For example:
1.- Open an existing form
2.- drop a new field
3.- Press rename
4.- Press save button
The form is renamed and saved but on reload the editor displays the previous form.

The rename & save operation on the service work well. The problem is that the form sent to the service has an old version of the layout. You need to synchronize the formLayout with the layout on the layoutEditor, try adding this on rename method before opening the popup:

```
if(save) {
    synchronizeFormLayout();
}
```

**Comment by pefernan** [comment]
jenkins execute full downstream build

**Comment by jsoltes** [comment]
jenkins retest this

**Comment by jsoltes** [comment]
@pefernan I believe now we only wait for your approval, pere...

**Pull request merged**